### PR TITLE
Connects frontend components to localStorage + jubSignal logic

### PR DESCRIPTION
**Pull request by AndrewCLu**
- [x] Populate activity feed and contacts list with localStorage data
- [x]  Connect users/[id] page with localStorage data
- [x] Update styling of users/[id]/share page
- [x] Make activity feed and contacts list click to user and location pages
- [x] Update timestamp to store as ISO strings

### CODE REVIEW

File: src/pages/users/[id]/share.tsx [hunk 1]
```diff
@@ -147,60 +161,48 @@ const SharePage = () => {
...
 
+  if (!profile) {
+    return <div>Loading...</div>;
+  }
+
   if (!user) {
     return <div>User not found</div>;
   }
 
   return (
```
Comment (RiverRuby) [inline src/pages/users/[id]/share.tsx:172]: We need frontends for this image.png

Comment (RiverRuby) [reply src/pages/users/[id]/share.tsx:172]: image

File: src/pages/users/[id]/share.tsx [hunk 2]
```diff
@@ -147,60 +161,48 @@ const SharePage = () => {
...
-      <div className="mt-8">
-        <button
-          className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
-          onClick={handleConnectClick}
-        >
-          Connect
-        </button>
-      </div>
+    <div>
+      <AppBackHeader />
```
Comment (RiverRuby) [inline src/pages/users/[id]/share.tsx:174]: Back button doesn't work here

File: src/pages/users/[id]/share.tsx [hunk 3]
```diff
@@ -147,60 +161,48 @@ const SharePage = () => {
...
+        />
+        <Input
+          type="longtext"
+          label="Private Note"
+          value={privateNote}
+          onChange={(event) => {
+            setPrivateNote(event.target.value);
+          }}
+        />
+        <Button type="submit">Connect</Button>
```
Comment (RiverRuby) [inline src/pages/users/[id]/share.tsx:204]: Not able to send data, not sure what my error is

File: src/pages/users/[id]/index.tsx [hunk 4]
```diff
@@ -19,56 +26,62 @@ const UserProfilePage = () => {
...
+          <div className="h-32 w-32 rounded bg-slate-200"></div>
+          <div className="flex flex-col gap-1">
+            <h2 className=" text-xl font-gray-12 font-light">{user.name}</h2>
+            <div className="flex items-center gap-1">
+              <Icons.checkedCircle />
+              <span className="text-sm font-light text-white">
+                Connected XXX
+              </span>
             </div>
+            <Button onClick={() => router.push(`/users/${id}/share`)}>
```
Comment (RiverRuby) [inline src/pages/users/[id]/index.tsx:42]: This should only appear if you haven't already connected right? Not by default?

File: src/pages/users/[id]/index.tsx [hunk 5]
```diff
@@ -19,56 +26,62 @@ const UserProfilePage = () => {
...
             {user.x && (
-              <div className="bg-gray-50 dark:bg-gray-900 px-4 py-5 grid grid-cols-1 gap-4 sm:grid-cols-3">
-                <dt className="text-sm font-medium text-gray-500 dark:text-gray-400">
-                  Twitter
-                </dt>
-                <dd className="mt-1 text-sm text-gray-900 dark:text-white sm:col-span-2">
-                  @{user.x}
-                </dd>
-              </div>
+              <Link href={`https://x.com/${user.x}`}>
```
Comment (RiverRuby) [inline src/pages/users/[id]/index.tsx:66]: We should make the input field during registration/profile update for twitter not say "twitter.com/username" it should start with an "@" and then just have a prompt for user name so this actually works.

File: src/pages/users/[id]/share.tsx [hunk 6]
```diff
@@ -147,60 +161,48 @@ const SharePage = () => {
...
+      <AppBackHeader />
+      <FormStepLayout
+        title={`Connect with ${user.name}`}
+        description="Share your contact information with this user."
+        onSubmit={handleConnect}
+      >
+        <Input
+          type="checkbox"
+          className="form-checkbox"
+          label={`Share my Twitter: @${profile.twitterUsername}`}
```
Comment (RiverRuby) [inline src/pages/users/[id]/share.tsx:183]: Definitely confusing that this alternates betwen `twitterUsername` and `x` for different objects

File: src/pages/locations/[id].tsx [hunk 7]
```diff
@@ -53,30 +54,33 @@ const LocationDetails = () => {
...
-          }}
-        >
-          <img src={location.imageUrl} />
-        </div>
-        <div className="flex flex-col gap-4 jus">
-          <div className="flex flex-col">
-            <Label>Description</Label>
-            <Description>{location.description}</Description>
+    <div>
+      <AppBackHeader />
```
Comment (RiverRuby) [inline src/pages/locations/[id].tsx:58]: This leads to an infinite loop where it takes you back to the tap page and it says you've already visited. This should link directly to social tab

File: src/pages/index.tsx [hunk 8]
```diff
@@ -41,7 +36,78 @@ const ContactCard = ({ name, date }: ContactCardProps) => {
...
+                    <u>{name}</u> {"connected with"} <u>You</u>
+                  </div>
+                }
+              </Card.Title>
+            </div>
+            <Card.Description>{date}</Card.Description>
+          </div>
+        </Link>
+      );
+    case JUB_SIGNAL_MESSAGE_TYPE.LOCATION_TAP:
```
Comment (RiverRuby) [inline src/pages/index.tsx:88]: Location tap isn't showing up in my feed

Comment (RiverRuby) [reply src/pages/index.tsx:88]: jk upon login it appears but it's duplicated 
image

**Review by RiverRuby — COMMENTED** [review]
I think there's a few bugs that need to be resolved before approval.

File: src/pages/users/[id]/share.tsx (hunk 6 above)
Comment (AndrewCLu) [reply src/pages/users/[id]/share.tsx:183]: The rule is x is used whenever space matters and twitterUsername is used everywhere else, to be more descriptive.

File: src/pages/users/[id]/index.tsx (hunk 4 above)
Comment (AndrewCLu) [reply src/pages/users/[id]/index.tsx:42]: I think this is a design discussion - i'm thinking its totally valid for people to reshare if they want to say update the information they send to someone, but we could also disable this

File: src/pages/users/[id]/share.tsx (hunk 3 above)
Comment (AndrewCLu) [reply src/pages/users/[id]/share.tsx:209]: Error was due to not calling `event.preventDefault()` on the form, oops...

File: src/pages/locations/[id].tsx (hunk 7 above)
Comment (AndrewCLu) [reply src/pages/locations/[id].tsx:58]: https://github.com/nfc-denver/nfc-denver/issues/68

File: src/pages/index.tsx (hunk 8 above)
Comment (AndrewCLu) [reply src/pages/index.tsx:88]: This is due to React double-triggering effects in dev environment on strict mode. We can either leave this (it has no effect on app logic) or add a ref that addresses this

File: src/pages/users/[id]/index.tsx (hunk 5 above)
Comment (AndrewCLu) [reply src/pages/users/[id]/index.tsx:66]: I think this is a good thing to bringup during the design review - we should definitely have a list of these small frontend fixes that kali can just knock out all at once

File: src/pages/users/[id]/share.tsx (hunk 1 above)
Comment (AndrewCLu) [reply src/pages/users/[id]/share.tsx:177]: https://github.com/nfc-denver/nfc-denver/issues/29

**Review by RiverRuby — APPROVED** [review]
LGTM, login for server back up is borked but i made an issue https://github.com/nfc-denver/nfc-denver/issues/70

**Review by RiverRuby — APPROVED** [review]
LGTM, login for server back up is borked but i made an issue https://github.com/nfc-denver/nfc-denver/issues/70

**Pull request merged**